include/librealsense2/dds/dds-device.cpp

```diff
@@ -36,7 +74,180 @@ class dds_device::impl
         , _participant( participant )
     {
         (void)guid;
+
+        create_notifications_reader();
+        if( !init() )
+        {
+            throw std::runtime_error("failed getting sensors data from device: " + info.topic_root);
+        }
+        LOG_DEBUG("Device" << _info.topic_root << " was initialize successfully" );
+    }
+
+private:
+
+    void create_notifications_reader()
+    {
+        // CREATE THE SUBSCRIBER
+        _subscriber = DDS_API_CALL( _participant->get()->create_subscriber( SUBSCRIBER_QOS_DEFAULT, nullptr ) );
+        if( _subscriber == nullptr )
+        {
+            throw std::runtime_error( "Error creating a DDS subscriber" );
+        }
+
+        // CREATE TOPIC TYPE
+        eprosima::fastdds::dds::TypeSupport topic_type(
+            new librealsense::dds::topics::device::notification::type );
+
+        // REGISTER THE TYPE
+        DDS_API_CALL( topic_type.register_type( _participant->get() ) );
+
+
+
+        std::string topic_name
+            = librealsense::dds::topics::device::notification::construct_topic_name( _info.topic_root );
+
+        // CREATE THE TOPIC
+        _topic = DDS_API_CALL( _participant->get()->create_topic( topic_name,
+                                                                  topic_type->getName(),
+                                                                  TOPIC_QOS_DEFAULT ) );
+
+        // CREATE THE READER
+        DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
+        rqos.data_sharing().off();
+        rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
+        rqos.durability().kind = VOLATILE_DURABILITY_QOS;
+        rqos.history().kind = KEEP_LAST_HISTORY_QOS;
+        rqos.history().depth = 10;
+        rqos.endpoint().history_memory_policy
+            = eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
+        _reader = DDS_API_CALL( _subscriber->create_datareader( _topic, rqos, nullptr ) );
+        if( _reader == nullptr )
+        {
+            throw std::runtime_error( "Error creating a DDS reader for " + topic_name );
+        }
+
+        LOG_DEBUG( "topic '" << topic_name << "' created" );
+    }
+
+    bool init()
+    {
+        size_t num_of_sensors = 0;
+        std::unordered_map< int , std::string> sensor_index_to_name;
+        std::unordered_map< std::string, topics::device::notification::video_stream_profiles_msg > sensor_to_video_profiles;
+        std::unordered_map< std::string, topics::device::notification::motion_stream_profiles_msg > sensor_to_motion_profiles;
+
+        // We expect to receive all of the sensors data under a timeout
+        utilities::time::timer t( std::chrono::seconds( 30) ); // TODO: refine time out
+        state_type state = state_type::WAIT_FOR_DEVICE_HEADER;
+
+        while( ! t.has_expired() && state_type::DONE != state )
+        {
+            if( _reader->wait_for_unread_message( { 1, 0 } ) )
```

Comment:
Since this is a fastrtps::Duration_t not std::duration I think it is best to create a named object and pass it as parameter
fastrtps::Duration_t one_second = { 1, 0  };
_reader->wait_for_unread_message( one_second );

Reply:
👍

Reply:
Done